src/dune_pkg/fetch.ml

```diff
@@ -56,16 +56,59 @@ module Curl = struct
            ([ Pp.textf "curl returned an invalid error code %d" exit_code ] @ stderr)))
     else (
       Path.unlink_no_err stderr;
-      match Int.of_string http_code with
+      (* Since we had to quote the return http_code, we need to strip it *)
+      match
+        let open Option.O in
+        String.drop_prefix_and_suffix ~prefix:"\"" ~suffix:"\"" http_code
+        >>= Int.of_string
+      with
       | None ->
         Error
           (User_message.make
              [ Pp.textf "curl returned an HTTP code we don't understand: %S" http_code ])
+      | Some 200 -> Ok ()
       | Some http_code ->
-        if http_code = 200
-        then Ok ()
-        else
-          Error (User_message.make [ Pp.textf "download failed with code %d" http_code ]))
+        Error (User_message.make [ Pp.textf "download failed with code %d" http_code ]))
+  ;;
+end
+
+module Tar = struct
+  let bin = lazy (Bin.which ~path:(Env_path.path Env.initial) "tar")
+
+  let run ~temp_dir input output =
+    let bin =
+      match Lazy.force bin with
+      | Some p -> p
+      | None -> User_error.raise [ Pp.text "tar not available in PATH" ]
+    in
+    let args =
+      [ "-xf"; Path.to_string input; "--force-local"; "-C"; Path.to_string output ]
+    in
+    let stderr = Path.relative temp_dir "tar.stderr" in
+    let+ _, exit_code =
+      let stderr_to = Process.Io.file stderr Out in
+      Process.run_capture_lines Return ~stderr_to ~display:Quiet bin args
```

Comment:
I wonder if it wouldn't be better to use `Strict` here, that way the whole validation and command output handling will be handled by `Process` in the same consistent way all other processes are handled.

Reply:
We use return in order to display any stderr we get from the process when we fail to extract. We do the same for curl too. I think giving the original command error in this case is more useful than a blanket Dune one, or even reconstructing the error.

Reply:
With quiet, we can't know what was outputted since we discarded it.